libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -262,14 +277,224 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
+    if (featureFlag.telemetry) {
+      featureFlagValue.telemetry = featureFlag.telemetry;
+    }
+
+    const clientFilters = featureFlag.conditions && featureFlag.conditions[this.featureFlagsKeyWordsCase.client_filters];
+    if (clientFilters && clientFilters.length != 0) {
+      featureFlagValue.conditions.clientFilters = this.validateClientFilters(featureFlagValue.id, "ffv2", clientFilters);
+    }
+
+    const requirementType = featureFlag.conditions && featureFlag.conditions[this.featureFlagsKeyWordsCase.requirement_type];
+    if (requirementType) {
+      if (this.validateRequirementType(requirementType, featureFlag.id)) {
+        featureFlagValue.conditions.requirement_type = requirementType;
+      }
+    }
+
+    if (featureFlag.allocation) {
+      featureFlagValue.allocation = {};
+      const default_when_disabled = featureFlag.allocation[this.featureFlagsKeyWordsCase.default_when_disabled];
+      if (default_when_disabled) {
+        featureFlagValue.allocation.default_when_disabled = default_when_disabled;
+      }
+
+      const default_when_enabled = featureFlag.allocation[this.featureFlagsKeyWordsCase.default_when_enabled];
+      if (default_when_enabled) {
+        featureFlagValue.allocation.default_when_enabled = default_when_enabled;
+      }
+
+      const userAllocation = this.getObjectWithKey(featureFlag.allocation, "user");
+      if (userAllocation) {
+        featureFlagValue.allocation.user = [];
+        for (const user of userAllocation) {
+          if (!this.isValidUserAllocation(user)) {
+            throw new ArgumentError(
+              `This feature flag '${featureFlag.id}' has a user allocation object without the required 'variant' and 'users' properties.`
+            );
+          }
+          featureFlagValue.allocation.user.push(user);
+        }
+      }
+
+      const groupAllocation = this.getObjectWithKey(featureFlag.allocation, "group");
+      if (groupAllocation) {
+        featureFlagValue.allocation.group = [];
+        for (const group of groupAllocation) {
+          if (!this.isValidGroupAllocation(group)) {
+            throw new ArgumentError(
+              `This feature flag '${featureFlag.id}' has a group allocation object without the required 'variant' and 'groups' properties.`
+            );
+          }
+          featureFlagValue.allocation.group.push(group);
+        }
+      }
+
+      const percentileAllocation = this.getObjectWithKey(featureFlag.allocation, "percentile");
+      if (percentileAllocation) {
+        featureFlagValue.allocation.percentile = [];
+        for (const percentile of percentileAllocation) {
+          if (!this.isValidPercentileAllocation(percentile)) {
+            throw new ArgumentError(
+              `This feature flag '${featureFlag.id}' has a percentile allocation object without the required 'from', 'to' and 'variant' properties.`
+            );
+          }
+          featureFlagValue.allocation.percentile.push(percentile);
```

Comment:
I think we should also validate the to and from values in the percentile object, minimum is `0` and maximum is `100`

Reply:
This is now validated by the ajv schema